lib/spack/spack/cmd/release_jobs.py

```diff
@@ -452,155 +369,263 @@ def append_dep(s, d):
+            'SPACK_CDASH_BASE_URL': cdash_url,
+            'SPACK_CDASH_PROJECT': cdash_project,
+            'SPACK_CDASH_PROJECT_ENC': cdash_project_enc,
+            'SPACK_CDASH_BUILD_NAME': cdash_build_name,
+            'SPACK_RELATED_BUILDS': ';'.join(related_builds),
+            'SPACK_ROOT_SPEC': root_spec,
+            'SPACK_JOB_SPEC_PKG_NAME': release_spec.name,
+            'SPACK_JOB_SPEC_BUILDGROUP': build_group,
+        }
+
+        if not compiler_job:
+            job_variables['SPACK_FIND_COMPILER'] = str(release_spec.compiler)
+
+        variables.update(job_variables)
+
+        job_object = {
+            'stage': stage_name,
+            'variables': variables,
+            'script': job_scripts,
+            'artifacts': {
+                'paths': [
+                    'local_mirror/build_cache',
+                    'jobs_scratch_dir',
+                    'cdash_report',
+                ],
+                'when': 'always',
+            },
+            'dependencies': job_dependencies,
+            'tags': tags,
+        }
+
+        if build_image:
+            job_object['image'] = build_image
+
+        output_object[job_name] = job_object
+
+        return True
+
+    phases = [
+        (compiler_stages, compiler_spec_labels, compiler_deps, True),
+        (stages, spec_labels, dependencies, False),
+    ]
 
-            if args.signing_key:
-                job_object['variables']['SIGN_KEY_HASH'] = args.signing_key
+    for stage_list, labels, deps, compiler_job in phases:
+        for stage_jobs in stage_list:
+            stage_name = stage_names[stage]
 
-            if do_job:
-                output_object[job_name] = job_object
-                job_count += 1
+            for spec_label in stage_jobs:
+                if add_job(spec_label, stage_name, labels, deps, compiler_job):
+                    job_count += 1
 
-        stage += 1
+            stage += 1
 
     tty.msg('{0} build jobs generated in {1} stages'.format(
-        job_count, len(stages)))
+        job_count, total_num_stages))
+
+    # Use "all_job_names" to populate the build group for this set
+    if cdash_auth_token:
+        try:
+            populate_buildgroup(all_job_names, build_group, cdash_project,
+                                cdash_site, cdash_auth_token, cdash_url)
+        except (SpackError, HTTPError, URLError) as err:
+            tty.warn('Problem populating buildgroup: {0}'.format(err))
+    else:
+        tty.warn('Unable to populate buildgroup without CDash credentials')
 
+    # Add an extra, final job to regenerate the index
     final_stage = 'stage-rebuild-index'
-
     final_job = {
         'stage': final_stage,
         'variables': {
-            'MIRROR_URL': mirror_url,
+            'MIRROR_URL': mirror_urls[0],
         },
-        'image': build_image,
+        'image': 'scottwittenburg/spack_ci_generator_alpine',  # just needs some basic python image
```

Comment:
I may be incorrect but this seems like it should be configuration, along with tags/variables?

Reply:
You're right, this final stage needs more attention, and stuff like that should be configurable ultimately.